quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusValidator.java

```diff
@@ -130,13 +130,61 @@ private void validatePropertyValue(String propertyName, ExtendedConfigDescriptio
 			return;
 		}
 
-		// TODO validate boolean, int, enums types
-		String type = metadata.getType();
+		String value = property.getValue().getText().trim();
+		if (value.length() == 0) {
+			return;
+		}
+		
+		String errorMessage = null;
+		if (isInvalidEnumValue(metadata, value)) {
+			errorMessage = "Invalid enum value: '" + property.getValue().getText().trim() + "' is invalid for type " + metadata.getType();
+		} else if (isValueTypeMismatch(metadata, value)) {
+			errorMessage = "Type mismatch: " + metadata.getType() + " expected";
+		}
+
+		if (errorMessage != null) {
+			addDiagnostic(errorMessage, property.getValue(), severity, ValidationType.value.name());
+		}
+	}
+
+	private boolean isInvalidEnumValue(ExtendedConfigDescriptionBuildItem metadata, String value) {

```

Comment:
Once https://github.com/redhat-developer/quarkus-ls/pull/100 will be accepted, remove this method and use `metadata.getEnumItem(value) != null`

Reply:
I think we still need `isInvalidEnumValue(metadata, value)`, because it also checks if the value is valid enum for the metadata or not